File: app/src/main/java/com/numero/sojodia/di/RepositoryModule.java
package com.numero.sojodia.di;

import android.content.Context;

import com.numero.sojodia.repository.BusDataRepository;
import com.numero.sojodia.repository.ConfigRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    public BusDataRepository provideBusDataRepository(Context context) {
        return new BusDataRepository(context);
    }

    @Provides
    @Singleton
    public ConfigRepository provideConfigRepository(Context context) {
        return new ConfigRepository(context);
    }
}

File: app/src/main/kotlin/com/numero/sojodia/di/RepositoryModule.kt
package com.numero.sojodia.di

import android.content.Context
import com.numero.sojodia.api.BusDataApi2
import com.numero.sojodia.repository.BusDataRepository
import com.numero.sojodia.repository.ConfigRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RepositoryModule {

    @Provides
    @Singleton
    fun provideBusDataRepository(context: Context, busDataApi2: BusDataApi2): BusDataRepository = BusDataRepository(context, busDataApi2)

    @Provides
    @Singleton
    fun provideConfigRepository(context: Context): ConfigRepository = ConfigRepository(context)
}
